browser/src/map/handler/Map.SlideShow.js

```diff
@@ -52,37 +52,45 @@ L.Map.SlideShow = L.Handler.extend({
 			return;
 		}
 
-		if (!this._cypressSVGPresentationTest && !this._map['wopi'].DownloadAsPostMessage) {
-			this._slideShow = L.DomUtil.create('iframe', 'leaflet-slideshow', this._map._container);
-			if (this._slideShow.requestFullscreen) {
-				this._slideShow.requestFullscreen();
-			}
-			else if (this._slideShow.msRequestFullscreen) {
-				this._slideShow.msRequestFullscreen();
-			}
-			else if (this._slideShow.mozRequestFullScreen) {
-				this._slideShow.mozRequestFullScreen();
+		let doPresentation = function(that, e) {
+			that._presentInWindow = false;
+
+			that._startSlideNumber = 0; // Default: start from page 0
+			if (typeof e.startSlideNumber !== 'undefined') {
+				that._startSlideNumber = e.startSlideNumber;
 			}
-			else if (this._slideShow.webkitRequestFullscreen) {
-				this._slideShow.webkitRequestFullscreen();
-			} else {
-				// fallback to "open in new tab"
-				L.DomUtil.remove(this._slideShow);
-				this._slideShow = null;
+			that.fullscreen = !that._cypressSVGPresentationTest;
+			that._map.downloadAs('slideshow.svg', 'svg', null, 'slideshow');
+
+			L.DomEvent.on(document, 'fullscreenchange', that._onFullScreenChange, that);
+		};
+
+		let fallback = function(that, e) {
+			// fallback to "open in new tab"
+			if (that._slideShow) {
+				L.DomUtil.remove(that._slideShow);
+				that._slideShow = null;
 			}
 
-			L.DomEvent.on(document, 'fullscreenchange webkitfullscreenchange mozfullscreenchange msfullscreenchange',
-				this._onFullScreenChange, this);
-		}
+			doPresentation(that, e);
+		};
 
-		this._presentInWindow = false;
+		if (!(this._cypressSVGPresentationTest || this._map['wopi'].DownloadAsPostMessage)) {
+			this._slideShow = L.DomUtil.create('iframe', 'leaflet-slideshow', this._map._container);
+			if (this._slideShow.requestFullscreen) {
 
-		this._startSlideNumber = 0; // Default: start from page 0
-		if (e.startSlideNumber !== undefined) {
-			this._startSlideNumber = e.startSlideNumber;
+				let that = this;
+				this._slideShow.requestFullscreen()
+					.then(function () { doPresentation(that, e); return true;})
```

Comment:
I don't think you need to return anything here.

Reply:
I had a warning, because the return of the `then` is used.

Reply:
it was because you passed the function result as a parameter expecting a function.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

It definitely return `undefined` in their example.

Reply:
```suggestion
					.then(function () { doPresentation(that, e); })
```